src/compiler/emitter.ts

```diff
@@ -2138,6 +2138,72 @@ module ts {
             function getTemplateLiteralAsStringLiteral(node: LiteralExpression): string {
                 return '"' + escapeString(node.text) + '"';
             }
+            
+            function emitDownlevelRawTemplateLiteral(node: LiteralExpression) {
+                var text = getSourceTextOfNodeFromSourceFile(currentSourceFile, node);
+                
+                // text contains the original source, it will also contain quotes ("`"), dolar signs and braces ("${" and "}"),
+                // thus we need to remove those characters.
+                // First template piece starts with "`", others with "}"
+                // Last template piece ends with "`", others with "${"
+                var isLast = node.kind === SyntaxKind.NoSubstitutionTemplateLiteral || node.kind === SyntaxKind.TemplateTail;
+                text = text.substring(1, text.length - (isLast ? 1 : 2));
+                
+                // Newline normalization
+                text = text.replace(/\r\n?/g, "\n");
+                text = escapeString(text);
+                
+                write('"' + text + '"');
+            }
+            
+            function emitDownlevelTaggedTemplate(node: TaggedTemplateExpression) {
+                var tempVariable = createTempVariable(node);
+                recordTempDeclaration(tempVariable);
+                write("(");
+                emit(tempVariable);
+                write(" = [");
+                
+                if (node.template.kind === SyntaxKind.NoSubstitutionTemplateLiteral) {
+                    emit(node.template);
+                }
+                else {
+                    emit((<TemplateExpression>node.template).head);
+                    forEach((<TemplateExpression>node.template).templateSpans, (child) => {
+                        write(", ");
+                        emit(child.literal);
+                    });
+                }
+                write("], ");
+                
+                emit(tempVariable);
+                write(".raw = [");
+                if (node.template.kind === SyntaxKind.NoSubstitutionTemplateLiteral) {
+                    emitDownlevelRawTemplateLiteral(<LiteralExpression>node.template);
+                }
+                else {
+                    emitDownlevelRawTemplateLiteral((<TemplateExpression>node.template).head);
+                    forEach((<TemplateExpression>node.template).templateSpans, (child, index) => {
+                        write(", ");
+                        emitDownlevelRawTemplateLiteral(child.literal);
+                    });
+                }
+                write("], ");
+                
+                emitParenthesizedIf(node.tag, needsParenthesisForPropertyAccessOrInvocation(node.tag));

```

Comment:
My point is that rather than each parent node needing to know that its child could be a template, and therefore needs to parenthesize that template, the template should parenthesize itself accordingly, because *it* is the one that lowers its precedence when it gets emitted.